src/include/miopen/solver.hpp

```diff
@@ -1538,6 +1538,42 @@ struct ConvBinWinogradRxSFused : SolverBase<ConvolutionContext>
     ConvSolution GetSolution(const ConvolutionContext& params) const;
 };
 
+template <int WinoDataH, int WinoFilterH, int WinoDataW = WinoDataH, int WinoFilterW = WinoFilterH>
+struct ConvMPBidirectWinograd : SolverBase<ConvolutionContext>
+{
+    bool IsApplicable(const ConvolutionContext& params) const;
+    size_t GetWorkspaceSize(const ConvolutionContext& params) const;
+    ConvSolution GetSolution(const ConvolutionContext& params) const;
+
+    // kernel_file_name for solver identification
+    static std::string GetSolverFileNames(int id)
+    {
+        static const std::string names[3] = {"xform_bidirect_winograd_data.s",
+                                             "xform_bidirect_winograd_filter.s",
+                                             "xform_bidirect_winograd_out.s"};
+        return names[id];
+    }
+
+    static std::string GetSolverKernelNames(int id)
+    {
+        static const std::string name_suffix =
+            '_' + std::to_string(WinoDataH) + '_' + std::to_string(WinoDataW) + '_' +
+            std::to_string(WinoFilterH) + '_' + std::to_string(WinoFilterW);
+        static const std::string names[3] = {
+            "miopenGcnAsmMPBidirectWinogradXformData" + name_suffix,
+            "miopenGcnAsmMPBidirectWinogradXformFilter" + name_suffix,
```

Comment:
Will the fwd and bwd filter transforms have the same name? Is it ok to have the same kernel name for different kernel binaries? Will the library work properly in that case?

Reply:
Yes, same.
Yes, as with most tunable solvers.
Yes, it works fine.